ui/appui-react/src/appui-react/childwindow/InternalChildWindowManager.tsx

```diff
@@ -92,8 +92,16 @@ export class InternalChildWindowManager implements FrameworkChildWindows {
     element: React.FunctionComponentElement<any>,
     container: Element | DocumentFragment
   ) {
+    // If createRoot is passed in for React 18 we have to render differently
+    // than without it. React 17 vs React 18. We need to save the root to
+    // unmount it.
     if (this._createRoot) {
-      this._createRoot(container).render(element);
+      this.openChildWindows.forEach((openChildWindow) => {
+        if (this._createRoot) {
+          openChildWindow.root = this._createRoot(container);
+          openChildWindow.root.render(element);
+        }
+      });
```

Comment:
With a `forEach`, we are replacing every windows roots with the latest one, so closing any window will actually unmount the latest window only ?
1. Pop widget A
2. Pop widget B
3. Close widget A
   -- I expect here that widget A will close, but widget B content will unmount.

Reply:
Implemented change